pygmt/tests/test_grdlandmask.py

```diff
@@ -49,7 +49,7 @@ def test_grdlandmask_no_outgrid(expected_grid):
     """
     Test grdlandmask with no set outgrid.
     """
-    result = grdlandmask(spacing=1, region=[125, 130, 30, 35])
+    result = grdlandmask(spacing=1, region=[125, 130, 30, 35], cores=1, verbose="i")
```

Comment:
Hmm, the debug info at https://github.com/GenericMappingTools/pygmt/actions/runs/7403072686/job/20142175894?pr=2944#step:8:111 is showing that there are 4 threads on GitHub Actions??

```
  grdlandmask [INFORMATION]: Enable 1 threads of 4 available
  grdlandmask [INFORMATION]: Selected ice front line as Antarctica coastline
  grdlandmask [INFORMATION]: GSHHG version 2.3.7
  grdlandmask [INFORMATION]: Derived from World Vector Shoreline, CIA WDB-II, and Atlas of the Cryosphere
  grdlandmask [INFORMATION]: Processed by Paul Wessel and Walter H. F. Smith, 1994-2017
  grdlandmask [INFORMATION]: Nodes in the oceans will be set to
  0
  grdlandmask [INFORMATION]: Nodes on land will be set to
  1
  grdlandmask [INFORMATION]: Nodes in lakes will be set to
  0
  grdlandmask [INFORMATION]: Nodes in islands will be set to
  1
  grdlandmask [INFORMATION]: Nodes in ponds will be set to
  0
  grdlandmask [INFORMATION]: Apply linear scale to avoid wrap-around: -Jx100id; this temporarily changes the domain
  grdlandmask [INFORMATION]: Writing grid to file /tmp/pygmt-0c0skk20.nc
  grdlandmask [INFORMATION]: Level 0 set for 33 nodes
  grdlandmask [INFORMATION]: Level 1 set for 3 nodes
  grdlandmask [INFORMATION]: Enable 1 threads of 4 available
  grdlandmask [INFORMATION]: Selected ice front line as Antarctica coastline
  grdlandmask [INFORMATION]: GSHHG version 2.3.7
  grdlandmask [INFORMATION]: Derived from World Vector Shoreline, CIA WDB-II, and Atlas of the Cryosphere
  grdlandmask [INFORMATION]: Processed by Paul Wessel and Walter H. F. Smith, 1994-2017
  grdlandmask [INFORMATION]: Nodes in the oceans will be set to
  0
  grdlandmask [INFORMATION]: Nodes on land will be set to
  1
  grdlandmask [INFORMATION]: Nodes in lakes will be set to
  0
  grdlandmask [INFORMATION]: Nodes in islands will be set to
  1
  grdlandmask [INFORMATION]: Nodes in ponds will be set to
  0
  grdlandmask [INFORMATION]: Apply linear scale to avoid wrap-around: -Jx100id; this temporarily changes the domain
  grdlandmask [INFORMATION]: Writing grid to file /tmp/pygmt-i3n6zkcg.nc
  grdlandmask [INFORMATION]: Level 0 set for 33 nodes
  grdlandmask [INFORMATION]: Level 1 set for 3 nodes
```

So the 2 CPUs mentioned at https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources might have hyperthreading, and we could potentially set `cores=4`, though see https://github.com/GenericMappingTools/pygmt/pull/2944#discussion_r1441172036 on how more cores might not actually improve runtime.